docs/content/src/learn/tutorials/integration-patterns-and-soa/backend-for-frontend/README.md

```diff
@@ -1,1187 +1,251 @@
+   ```bash
+   .
+   ├── Ballerina.toml
+   └── src
+      └── healthcare_service
+         ├── backends_for_frontends
+         ├── downstream_services
+         ├── Module.md
+         ├── resources
+         └── tests
+               └── resources
+   ```
 
-```bash
-   $ ballerina run appointment_mgt 
-```
+* Now that we have created the project structure, the next step is to develop the services.
 
-```bash
-   $ ballerina run medical_record_mgt
-```
+   First let’s implement the set of downstream services under `downstream_services` directory. Here we're going to create four services appointment_mgt_service, medical_record_mgt_service, message_mgt_service, and notification_mgt_service, each responding with pre-defined JSON payloads for demonstration purpose.
 
-```bash
-   $ ballerina run notification_mgt
-```
+**appointment_mgt_service.bal**
 
-```bash
-   $ ballerina run message_mgt
-```
-The commands start appointment_mgt_service, medical_record_mgt_service, notification_mgt_service and message_mgt_service on ports 9092, 9093, 9094 and 9095 respectively. 
+<!-- INCLUDE_CODE: src/healthcare_service/downstream_services/appointment_mgt_service.bal -->
 
-Similarly, execute the following commands via separate terminals to start the BFF layer services: 
+**medical_record_mgt_service.bal**
 
+<!-- INCLUDE_CODE: src/healthcare_service/downstream_services/medical_record_mgt_service.bal -->
 
-```bash
-   $ ballerina run mobile_bff 
-```
+**message_mgt_service.bal**
 
-```bash
-   $ ballerina run desktop_bff
-```
-The commands start mobile_bff_service and desktop_bff_service on ports 9090 and 9091 respectively. 
+<!-- INCLUDE_CODE: src/healthcare_service/downstream_services/message_mgt_service.bal -->
 
-For demonstration purpose let’s add sample data to the downstream services. Execute the following command to load sample appointments, medical records, notifications and messages to the services. 
+**notification_mgt_service.bal**
+<!-- INCLUDE_CODE: src/healthcare_service/downstream_services/notification_mgt_service.bal -->
 
-```bash
-   $ ballerina run sample_data_publisher
-```
+* Now let’s move to the key implementation of BFF services under `backends_for_frontends` directory. 
 
-Now that you have sample data loaded to the downstream services, you can call the BFF layer to retrieve the data based on the requirement. 
+The first BFF is the `mobile_bff_service` which is a shim used to support Mobile user experience in this use case. When loading mobile application's home page, it calls a single resource in Mobile BFF and retrieves appointments, medical records and messages. This will reduce the number of backend calls and will help to load the home pages in much efficient way. Since different mobile apps have different methods of sending notifications, we are not using the notification management service.
```

Comment:
```suggestion
The first BFF is the `mobile_bff_service`, which is a shim used to support Mobile user experience in this use case. When loading mobile application's home page, it calls a single resource in Mobile BFF and retrieves appointments, medical records, and messages. This will reduce the number of backend calls and will help to load the home pages in a more efficient way. Since different mobile apps have different methods of sending notifications, we are not using the notification management service.
```